src/common/AndroidEnvironmentRequirements.ts

```diff
@@ -77,7 +77,7 @@ export class Java8AvailableRequirement implements Requirement {
      * Verifies that prerequisites for Android SDK are met and that Java 8 is installed.
      */
     public async checkFunction(): Promise<string> {
-        return AndroidUtils.androidSDKPrerequisitesCheck()
+        return AndroidUtils.androidSDKPrerequisitesCheck(this.logger)
```

Comment:
This is something that I've always wanted to do but kept postponing. Rather than having `static` loggers in multiple places, we should just update method signatures to take an optional logger. I think this is cleaner. Not to mention that, we had a couple of utility classes with static loggers, but these loggers were not being initialized in `LoggerSetup.ts` which was a bug. Getting rid of these static loggers will also help us avoid such issues in the future .

Reply:
I don't know that I think it is cleaner. But I also think this is because our code base is largely functional, as opposed to OO with separations of concerns and responsibilities. In an OO world we'd create objects that hold onto common configuration such as this. But we're pretty far from that, so it's fair to embrace the functional nature of our project and take this approach. 👍